### PR TITLE
En_Kbt

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Kbt/z_en_kbt.c [hunk 1]
```diff
@@ -31,36 +34,543 @@ const ActorInit En_Kbt_InitVars = {
...
+    u8 sp38[] = {
+        ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_LOOP,
+        ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE,
+    };
+    static AnimationHeader* sAnimations[] = {
+        &object_kbt_Anim_000670, &object_kbt_Anim_001674, &object_kbt_Anim_002084, &object_kbt_Anim_000FE8,
+        &object_kbt_Anim_001940, &object_kbt_Anim_00E7BC, &object_kbt_Anim_00F0C8, &object_kbt_Anim_002710,
+        &object_kbt_Anim_002DE0, &object_kbt_Anim_003414, &object_kbt_Anim_003D24, &object_kbt_Anim_001BF4,
+        &object_kbt_Anim_002084,
+    };
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:95]: I prefer in-function static to be above the nonstatics.

File: src/overlays/actors/ovl_En_Kbt/z_en_kbt.c [hunk 2]
```diff
@@ -31,36 +34,543 @@ const ActorInit En_Kbt_InitVars = {
...
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+
+    gfx = POLY_OPA_DISP;
+
+    temp = Lib_SegmentedToVirtual(D_80B34B98[this->unk_27E]);
+    gSPSegment(&gfx[0], 0x08, temp);
+
+    temp = Lib_SegmentedToVirtual(D_80B34B90[this->unk_27F]);
+    gSPSegment(&gfx[1], 0x09, temp);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:568]: Is this temp required, or is it just for stack? There should be room for a pad here.

Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:568]: It's required for codegen.

File: src/overlays/actors/ovl_En_Kbt/z_en_kbt.c [hunk 3]
```diff
@@ -31,36 +34,543 @@ const ActorInit En_Kbt_InitVars = {
...
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kbt/func_80B3403C.s")
+Actor* func_80B3403C(GlobalContext* globalCtx) {
+    Actor* npc = globalCtx->actorCtx.actorLists[ACTORCAT_NPC].first;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kbt/func_80B34078.s")
+    while (npc != NULL) {
+        if (npc->id == 0x1FF) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:113]:
```suggestion
        if (npc->id == ACTOR_EN_KGY) {
```

File: src/overlays/actors/ovl_En_Kbt/z_en_kbt.c [hunk 4]
```diff
@@ -31,36 +34,543 @@ const ActorInit En_Kbt_InitVars = {
...
+        this->unk_278->textId = this->actor.textId;
+        this->unk_27C &= ~4;
+        if (this->actor.textId == 0xC4E) {
+            this->unk_27C |= 0x10;
+        }
+    } else if (this->actor.xzDistToPlayer < 250.0f) {
+        if ((this->unk_278 != NULL) && (this->unk_278->xzDistToPlayer < 250.0f)) {
+            if (this->unk_27C & 4) {
+                playerForm = gSaveContext.playerForm;
+                if (((playerForm ^ 0) != 4) || ((CUR_FORM_EQUIP(EQUIP_SLOT_B) != ITEM_SWORD_KOKIRI) &&
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:254]:
```suggestion
                if (((playerForm ^ 0) != PLAYER_FORM_HUMAN) || ((CUR_FORM_EQUIP(EQUIP_SLOT_B) != ITEM_SWORD_KOKIRI) &&
```

File: src/overlays/actors/ovl_En_Kbt/z_en_kbt.c [hunk 5]
```diff
@@ -31,36 +34,543 @@ const ActorInit En_Kbt_InitVars = {
...
+};
+
+TexturePtr D_80B34B98[] = {
+    object_kbt_Tex_00B5B8, object_kbt_Tex_00BDB8, object_kbt_Tex_00C5B8, object_kbt_Tex_00CDB8, object_kbt_Tex_00D5B8,
+};
+
+void EnKbt_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnKbt* this = THIS;
+    Gfx* gfx;
+    void* temp;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:556]:
```suggestion
    TexutrePtr tex;
```
Used as a ptr to a texture

**Review by hensldm — COMMENTED** [review]
A few things

File: src/overlays/actors/ovl_En_Kbt/z_en_kbt.c (hunk 1 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:95]: I put it in data order, I didn't realise I could swap them.

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Kbt/z_en_kbt.c:95]: Oh, huh, didn't notice only one was... probably stuck with it as-is, then.

**Review by hensldm — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Pull request merged**